### PR TITLE
Sgandon/tcomp 557 fix salesforce api for esb

### PR DESCRIPTION
**Pull request by sgandon**
**What is the current behavior?** (You can also link to an open issue here)
ESB fail to use the SalesforceSourceOrSing static APIs since the split between definitions and runtime

**What is the new behavior?**
Sandbox used have been removed from those runtime class, cause there is not reason.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/386/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/386/)

File: components/components-salesforce/components-salesforce-integration/src/test/java/org/talend/components/salesforce/runtime/SalesforceWriterTestIT.java [hunk 1]
```diff
@@ -579,7 +580,8 @@ public void testSinkWorkflow_insertRejected() throws Exception {
         assertThat(rejected.getSchema().getFields().get(6).name(), is("errorMessage"));
         assertThat(rejected.get(4), is((Object) "REQUIRED_FIELD_MISSING"));
         assertThat(rejected.get(5), is((Object) "Name"));
-        assertThat(rejected.get(6), is((Object) "Required fields are missing: [Name]"));
+        // removed the check on value cause it is i18n
```
Comment (jzhao-talend) [inline components/components-salesforce/components-salesforce-integration/src/test/java/org/talend/components/salesforce/runtime/SalesforceWriterTestIT.java]: I have a question about i18 issue here. This error message is return from server side. It is not define by us.
image
It's language is depend on server side setting. Do you think it's possible to add i18n  in our side ?

File: components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceSourceOrSink.java [hunk 2]
```diff
@@ -338,20 +324,15 @@ public SessionRenewalHeader renewSession(ConnectorConfig connectorConfig) throws
...
-            SalesforceSourceOrSink ss = (SalesforceSourceOrSink) sandboxedInstance.getInstance();
-            ss.initialize(null, (ComponentProperties) properties);
-            PartnerConnection connection = null;
-            try {
-                connection = ss.connect(container).connection;
-            } catch (IOException ex) {
-                throw new ComponentException(SalesforceRuntimeCommon.exceptionToValidationResult(ex));
-            }
-            return ss.getSchema(connection, module);
+        SalesforceSourceOrSink ss = new SalesforceSourceOrSink();
```
Comment (jzhao-talend) [inline components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceSourceOrSink.java]: I found this kind modification would remove the modification in Talend/components/pull/259
I am not sure whether  this caused that issue again. Could you explain me why this part of code revert back?

Comment (sgandon) [inline components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceSourceOrSink.java]: You are right my friend, I forgot about that. I'll try to fix this.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/390/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/390/)

**Pull request merged**